Update the `search-replace` command to follow the VIP-CLI style guide

yolih opened the pull request:
## Description

Updated the voice, tense, and format of command descriptions, examples, and usage models for the `vip search-replace` command to follow the VIP-CLI style guide.

## Pull request checklist

- [ n/a ] Update [SETUP.md](https://github.com/Automattic/vip-cli/blob/trunk/docs/SETUP.md#list-of-environmental-variables) with any new environmental variables.
- [ ] Update [the documentation](https://github.com/Automattic/vip-cli/blob/trunk/docs).
- [ ] [Manually test](https://github.com/Automattic/vip-cli/blob/trunk/docs/TESTING.md#manual-testing) the relevant changes.
- [ x ] Follow the [pull request checklist](https://github.com/Automattic/vip-cli/blob/trunk/docs/RELEASING.md#new-pull-requests)
- [ x ] Add/update [automated tests](https://github.com/Automattic/vip-cli/blob/trunk/docs/TESTING.md#automated-testing) as needed.

## New release checklist

- [ ] [Automated tests](https://github.com/Automattic/vip-cli/blob/trunk/docs/TESTING.md#automated-testing) pass.
- [ ] The [Preparing for release checklist](https://github.com/Automattic/vip-cli/blob/trunk/docs/RELEASING.md#preparing-for-release) is completed.

## Steps to Test

Observe the results of the `--help` display for the impacted commands:

- `vip`
- `vip search-replace`

sjinks committed:
fix(dev-env): verify the container is running before URL scan

yolih committed:
Updates to descriptions and examples

dependabot[bot] committed:
build(deps-dev): bump @types/node from 22.8.5 to 22.8.6

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.8.5 to 22.8.6.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sjinks commented on src/bin/vip-search-replace.js:
```diff
@@ -12,33 +12,39 @@ const debug = debugLib( '@automattic/vip:bin:vip-search-replace' );
-		description: 'Perform Search and Replace explicitly on the provided input <file.sql> file',
+		usage: 'vip search-replace file.sql --search-replace="from,to" --in-place',
+		description:
+			'Perform the search and replace operation and save the results to the local input file "file.sql".',
```
Should we mention that the S&R operation is performed on `file.sql`?

yolih replied:
I'm having a hard time understanding what needs to be improved in order to address your concern.

Is this wording a bit better?
`Perform the search and replace operation on the local input file "file.sql" and overwrite the file with the results.`

If not, I welcome a version of the description from you that is clearer. 🙏

sjinks commented on src/bin/vip-search-replace.js:
```diff
@@ -12,33 +12,39 @@ const debug = debugLib( '@automattic/vip:bin:vip-search-replace' );
+		usage: 'vip search-replace file.sql --search-replace="from,to" --output=output-file.sql',
 		description:
-			'Search and Replace to the specified output <output.sql> file\n' +
-			'       * Has no effect when the `in-place` flag is used',
+			'Perform the search and replace operation and save the results to a local clone of the input file named "output-file.sql".',
```
It is unclear to me (from the previous description) where the output will go. Given the "Has no effect when the `in-place` flag is used," I would assume that the in-place operation would be performed on the source file (`file.sql`) and the result will be saved to that very file, not to `output-file.sql`.

sjinks replied:
See https://github.com/Automattic/vip-cli/pull/2079/files#diff-1d697c6fa0eb15c3a87ec97840846b8674b75621357aac3c1e65681f03f25708R47

yolih replied:
One of the goals of the updated descriptions is to put any limitations or requirements for a command option in the description for the option, not in the usage example. 

I agree with you that the "Has no effect" phrase is unclear. What if we updated the description for `--output` to something like this:

```
.option(
		'output',
		'Clone the local input file and overwrite the clone with the results of the search and replace operation. Accepts a local file path. Overridden by --in-place if both options are run in the same command.'
	)
```

_(I edited the above description after reading the suggestion from @aswasif007)_

terriann replied:
I'd suggest instead:

```
A local file name to save the results of the search and replace operation. Has no effect if the --in-place parameter is also included in the command.
```

yolih replied:
I think it's important to make it clear that the option will create the file; that the file does not need to pre-exist for the option to work. 🤔

sjinks commented on src/bin/vip-search-replace.js:
```diff
@@ -12,33 +12,39 @@ const debug = debugLib( '@automattic/vip:bin:vip-search-replace' );
-	.option( 'in-place', 'Perform the search and replace explicitly on the input file' )
+	.option(
+		'search-replace',
+		'Pass a string value to search for and a string value to replace it with. Separate the values by a comma only; no spaces (e.g. --search-replace=“from,to”).'
```
```suggestion
		'Pass a string value to search for and a string value to replace it with. Separate the values by a comma only; no spaces (e.g. --search-replace="from,to").'
```

(fixed quotes in case someone copies and pastes the example)

aswasif007 commented on src/bin/vip-search-replace.js:
```diff
@@ -12,33 +12,39 @@ const debug = debugLib( '@automattic/vip:bin:vip-search-replace' );
+	)
+	.option(
+		'in-place',
+		'Save the results of a search and replace operation to the local input file.'
```
I feel like we should use the word "overwrite" somewhere in there, because that is more catchy, and as soon as the user notices this word, the purpose of `in-place` becomes clear without reading the full sentence.

Example:
`Overwrite the local input file with the results of the search and replace operation`

yolih replied:
I like that suggestion!

aswasif007 commented on src/bin/vip-import-sql.js:
```diff
@@ -400,7 +400,7 @@ void command( {
 	.option(
 		'in-place',
-		'Perform a search and replace operation on a local SQL file, save the results to the file, then import the updated file.'
+		'Perform a search and replace operation on the local input file, and save the results to the file prior to import.'
```
Same as https://github.com/Automattic/vip-cli/pull/2079#discussion_r1826916772

yolih replied:
How about this:

```
.option(
		'in-place',
		'Overwrite the local input file with the results of the search and replace operation prior to import.'
	)
	.option(
		'output',
		'Clone the local input file and overwrite the clone with the results of the search and replace operation. Accepts a local file path. Overridden by --in-place if both options are run in the same command, and ignored if the command excludes a --search-replace operation.'
	)
```

aswasif007 replied:
That looks good! 👍

dependabot[bot] committed:
build(deps-dev): bump @types/node from 22.8.6 to 22.8.7

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.8.6 to 22.8.7.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sjinks committed:
Update src/lib/dev-environment/dev-environment-lando.ts

Co-authored-by: Rinat K <rinat.khaziev@automattic.com>

yolih committed:
Update src/bin/vip-search-replace.js

Co-authored-by: Volodymyr Kolesnykov <volodymyr.kolesnykov@automattic.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 22.8.7 to 22.9.0

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.8.7 to 22.9.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

terriann commented on src/bin/vip-search-replace.js:
```diff
@@ -12,33 +12,39 @@ const debug = debugLib( '@automattic/vip:bin:vip-search-replace' );
-	.option( 'in-place', 'Perform the search and replace explicitly on the input file' )
+	.option(
+		'search-replace',
+		'Pass a string value to search for and a string value to replace it with. Separate the values by a comma only; no spaces (e.g. --search-replace="from,to").'
```
We can make this more concise.

```suggestion
		'A comma-separated pair of strings specifying the search and replacement values (e.g., --search-replace="from,to").'
```

yolih replied:
I like your proposed concise version! I am inclined to stick with the term "replace" instead of "replacement" though. What do you think of this version?
`A comma-separated pair of strings that specify the values to search for and replace (e.g., --search-replace="from,to").`

sjinks committed:
fix(dev-env): detection of URLs to replace for multisites (#2081)

* fix(dev-env): detection of URLs to replace for multisites

* fix: replacement takes paths into consideration

* refactor: simplify `replaceDomain()`

* fix: update `wp_blogs`

* Fix test cases - we don't have id's anymore

---------

Co-authored-by: Rinat Khaziev <rinat.khaziev@gmail.com>

dependabot[bot] committed:
build(deps): bump uuid from 11.0.2 to 11.0.3

Bumps [uuid](https://github.com/uuidjs/uuid) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/uuidjs/uuid/releases)
- [Changelog](https://github.com/uuidjs/uuid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/uuidjs/uuid/compare/v11.0.2...v11.0.3)

---
updated-dependencies:
- dependency-name: uuid
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump nock from 13.5.5 to 13.5.6

Bumps [nock](https://github.com/nock/nock) from 13.5.5 to 13.5.6.
- [Release notes](https://github.com/nock/nock/releases)
- [Changelog](https://github.com/nock/nock/blob/main/CHANGELOG.md)
- [Commits](https://github.com/nock/nock/compare/v13.5.5...v13.5.6)

---
updated-dependencies:
- dependency-name: nock
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yolih committed:
Changes as per feedback in progress

yolih committed:
Updates as per feedback

yolih committed:
Merging trunk

terriann commented on src/bin/vip-search-replace.js:
```diff
@@ -12,33 +12,39 @@ const debug = debugLib( '@automattic/vip:bin:vip-search-replace' );
 	.option(
 		'output',
-		'Create a local copy of the file with the completed search and replace operations. Ignored if the command includes --in-place. Accepts a local file path. Defaults to STDOUT.'
+		'Clone the local input file and overwrite the clone with the results of the search and replace operation. Accepts a local file path. Overridden by --in-place if both options are run in the same command.'
```
This can be more concise and use more straightforward language without losing clarity.

```suggestion
		'The local file path used to save a copy of the results from the search and replace operation. Ignored when used with the --in-place option.'
```

Overridden feels inaccurate because the output location isn't superseded by a value passed with the `--in-place` argument; it's ignored.

terriann commented on src/bin/vip-import-sql.js:
```diff
@@ -400,11 +400,11 @@ void command( {
 	.option(
 		'output',
-		'Create a local copy of the imported file with the completed search and replace operations. Ignored if the command includes --in-place, or excludes a --search-replace operation. Accepts a local file path.'
+		'Clone the local input file and overwrite the clone with the results of the search and replace operation. Accepts a local file path. Overridden by --in-place if both options are run in the same command, and ignored if the command excludes a --search-replace operation.'
```
This description is very verbose. The following feels just as accurate in describing the outcome for the user while using fewer words and simpler language.

```suggestion
		'The local file path to save a copy of the results from the search and replace operation when the --search-replace option is passed. Ignored when used with the --in-place option.'
```

yolih committed:
Update src/bin/vip-import-sql.js

Co-authored-by: Terri Ann Swallow <terri.swallow@automattic.com>

yolih committed:
Update src/bin/vip-search-replace.js

Co-authored-by: Terri Ann Swallow <terri.swallow@automattic.com>

terriann reviewed (approved):
👍
